libs/api_common/registry_helper.cpp

```diff
@@ -0,0 +1,214 @@
+}
+
+uint32_t
+write_registry_value_ansi_string(ebpf_registry_key_t key, _In_z_ const wchar_t* value_name, _In_z_ const char* value)
+{
+    uint32_t result;
+    try {
+        auto wide_string = _get_wstring_from_string(value);
+        result = write_registry_value_wide_string(key, value_name, wide_string.c_str());
+    } catch (...) {
+        result = ERROR_NOT_ENOUGH_MEMORY;
+    }
+
+    return result;
+}
+
+uint32_t
+write_registry_value_dword(ebpf_registry_key_t key, _In_z_ const wchar_t* value_name, uint32_t value)
+{
+    ebpf_assert(key);
+    return RegSetValueEx(key, value_name, 0, REG_DWORD, (PBYTE)&value, sizeof(value));
+}
+
+uint32_t
+create_registry_key(
+    ebpf_registry_key_t root_key, _In_ const wchar_t* sub_key, uint32_t flags, _Out_ ebpf_registry_key_t* key)
+{
+    *key = nullptr;
+    if (root_key == nullptr) {
+        return ERROR_INVALID_PARAMETER;
+    }
+
+    return RegCreateKeyEx(root_key, sub_key, 0, nullptr, 0, flags, nullptr, key, nullptr);
+}
+
+uint32_t
+open_registry_key(
+    ebpf_registry_key_t root_key, _In_opt_z_ const wchar_t* sub_key, uint32_t flags, _Out_ ebpf_registry_key_t* key)
+{
+    ebpf_assert(root_key != nullptr);
+
+    return RegOpenKeyEx(root_key, sub_key, 0, flags, key);
+}
+
+uint32_t
+delete_registry_key(ebpf_registry_key_t root_key, _In_z_ const wchar_t* sub_key)
+{
+    return RegDeleteKeyEx(root_key, sub_key, 0, 0);
+}
+
+uint32_t
+delete_registry_tree(ebpf_registry_key_t root_key, _In_opt_z_ const wchar_t* sub_key)
+{
+    return RegDeleteTree(root_key, sub_key);
+}
+
+uint32_t
+create_registry_key_ansi(
+    ebpf_registry_key_t root_key, _In_z_ const char* sub_key, uint32_t flags, _Out_ ebpf_registry_key_t* key)
+{
+    uint32_t result;
+    try {
+        auto wide_string = _get_wstring_from_string(sub_key);
+        result = create_registry_key(root_key, wide_string.c_str(), flags, key);
+    } catch (...) {
+        result = ERROR_NOT_ENOUGH_MEMORY;
+    }
+
+    return result;
+}
+
+uint32_t
+read_registry_value_string(ebpf_registry_key_t key, _In_ const wchar_t* value_name, _Outptr_result_z_ wchar_t** value)
+{
+    uint32_t status = NO_ERROR;
+    DWORD type = REG_SZ;
+    DWORD value_size = 0;
+    wchar_t* string_value = nullptr;
+
+    *value = nullptr;
+    status = RegQueryValueEx(key, value_name, 0, &type, nullptr, &value_size);
+    if (status != ERROR_SUCCESS || type != REG_SZ) {
+        return win32_error_code_to_ebpf_result(status);
+    }
+
+    string_value = (wchar_t*)ebpf_allocate((value_size + sizeof(wchar_t)));
+    if (string_value == nullptr) {
+        status = ERROR_NOT_ENOUGH_MEMORY;
+        return win32_error_code_to_ebpf_result(status);
+    }
+
+    status = RegQueryValueEx(key, value_name, 0, &type, (PBYTE)string_value, &value_size);
+    if (status != ERROR_SUCCESS) {
+        goto Exit;
+    }
+    *value = string_value;
+    string_value = nullptr;
+
+Exit:
+    if (string_value) {
+        ebpf_free(string_value);
+    }
+    return status;
```

Comment:
Other returns in the function above return `win32_error_code_to_ebpf_result(status)` instead of status, these should be consistent.
If you return ebpf_result_t then change the return type to that.